src/Components/Endpoints/src/RazorComponentEndpointInvoker.cs

```diff
@@ -116,6 +117,16 @@ private async Task RenderComponentCore(HttpContext context)
             }
         }
 
+        if (!htmlContent.QuiescenceTask.IsCompleted)
```

Comment:
Should this be using the `quiesceTask` local instead?
```suggestion
        if (!quiesceTask.IsCompleted)
```
That way it handles the form post case as well.